### PR TITLE
feat(bin) enhance logging config to make RUST_LOG for journalId and File writers

### LINKED ISSUE #5685: Journald and file log writers ignore RUST_LOG environment variable
### Describe the bug

In this function, `EnvFilter` is only instantiated with default and config filters, while `RUST_LOG` only affects the stdout.

I think `RUST_LOG` should be additive to the existing filters for these layers

https://github.com/paradigmxyz/reth/blob/a100fb3e46cb5f40d6580ece884effacae37b0c7/bin/reth/src/cli/mod.rs#L228

### Steps to reproduce

1. `RUST_LOG=info,rpc=trace reth node ...`
2. stdout/console will display `rpc::*` `TRACE` logs
3. file in `~/.cache/reth/logs/<chain>/reth.log` displays only up to `DEBUG`, and no `rpc` logs at all

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

73a5b6ced9fb71f485fe3ced97101cbb84bcf348

### What database version are you on?

_No response_

### What type of node are you running?

None

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
Resolves https://github.com/paradigmxyz/reth/issues/5685

### CODE REVIEW

**Comment by DaniPopes** [comment]
@DoTheBestToGetTheBest FYI to close issues automatically https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

**Review by shekhirin — CHANGES_REQUESTED** [review]
Hmm, can we initialize the `EnvFilter` with `from_env_lossy`, and then use `add_directive` to add `DEFAULT_ENV_FILTER_DIRECTIVE` and filter from arguments?

`add_directive` works by overwriting existing spans and events: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#method.add_directive

**Comment by DoTheBestToGetTheBest** [comment]
> nt / clippy / ethereum (pull_reque

ty for your review and your help, changed file :)

File: bin/reth/src/cli/mod.rs [hunk 1]
```diff
@@ -232,20 +232,27 @@ impl Logs {
     {
         let mut layers = Vec::new();
 
+        // Function to create a new EnvFilter with base and specific directives
+        let create_env_filter = |additional_directive: &str| -> eyre::Result<EnvFilter> {
+            let mut filter = EnvFilter::builder().from_env_lossy();
+            filter = filter.add_directive(DEFAULT_ENV_FILTER_DIRECTIVE.parse()?);
+            filter = filter.add_directive(additional_directive.parse()?);
+            Ok(filter)
```
Comment (shekhirin) [inline bin/reth/src/cli/mod.rs:240]: should work in one statement with builder pattern

```suggestion
            Ok(EnvFilter::builder().from_env_lossy()
                .add_directive(DEFAULT_ENV_FILTER_DIRECTIVE.parse()?)
                .add_directive(additional_directive.parse()?))
```

File: bin/reth/src/cli/mod.rs [hunk 2]
```diff
@@ -232,20 +232,27 @@ impl Logs {
     {
         let mut layers = Vec::new();
 
+        // Function to create a new EnvFilter with base and specific directives
```
Comment (shekhirin) [inline bin/reth/src/cli/mod.rs:235]:
```suggestion
        // Function to create a new EnvFilter with environment, default and additional directive
```

**Review by shekhirin — APPROVED** [review]
LGTM, only nits!

**Comment by DoTheBestToGetTheBest** [comment]
> LGTM, only nits!

ty so much  !

**Review by shekhirin — APPROVED** [review]
LGTM, thank you! Really clean change.

**Comment by DoTheBestToGetTheBest** [comment]
> LGTM, thank you! Really clean change.

ty sir, appreciate lot

**Pull request merged**